libs/flux/build.go

```diff
@@ -263,6 +269,31 @@ func downloadModule(logger *zap.Logger) (module.Version, string, error) {
 }
 
 func getVersion(dir string, logger *zap.Logger) (string, error) {
+	if v, err := getVersionFromPath(dir); err != nil {
+		logger.Info("Could not determine version from base path", zap.Error(err))
+	} else {
+		return v, nil
+	}
+
+	if v, err := getVersionFromGit(dir, logger); err != nil {
+		logger.Info("Could not determine version from git data", zap.Error(err))
+	} else {
+		return v, nil
+	}
+	logger.Info("Using default version")
+	return "", errors.New("unable to determine flux repository version")
+}
+
+func getVersionFromPath(dir string) (string, error) {
+	reModulePath := regexp.MustCompile(`/github\.com/influxdata\/flux@(v\d+\.\d+\.\d+.*)$`)
+	m := reModulePath.FindStringSubmatch(dir)
+	if m == nil {
+		return "", fmt.Errorf("directory path did not match the module pattern: %s", dir)
+	}
+	return m[1], nil
+}
+
+func getVersionFromGit(dir string, logger *zap.Logger) (string, error) {
```

Comment:
This will succeed if we are building from the Flux repo root (i.e., either we are building Flux, or we are building influxdb for example, but have used "mod replace" to point at a Flux repo on the local file system).  Right?

Reply:
Yes.